pkg/msg/handler.go

```diff
@@ -66,6 +66,7 @@ func (d *Dispatcher) readLoop() {
 		m, err := ReadMsg(d.rw)
 		if err != nil {
 			close(d.doneCh)
+			close(d.sendCh)
```

Comment:
sendCh does not need to be actively closed; it should be automatically reclaimed after all senders have exited.

As I commented in transport/message.go, I prefer to use a reasonable mechanism to ensure exit on the sender's side.